woocommerce/woocommerce
# Revert improved filtering for variations

### All Submissions:


### Changes proposed in this Pull Request:

This pull request reverts the functionality introduced in #26260 (later refined by #27175, #27190, #27508) in which filtering by
attribute using the layered nav widget was improved to handle the cases of variations out of stock. The revert is a response to the numerous problems related to filtering reported by users in WooCommerce 4.4 and 4.5, including two more that we hadn't detected previously:

* Attributes not used for variations aren't handled correctly.
* Products marked as "hidden for catalog" are still being counted.

We'll figure out a better way to achieve the same functionality and re-implement the feature in a future version of WooCommerce.

Not all the code has been reverted, only the code that resulted in visible functionality changes. Thus, the code that generates
term relationships for variations is still in place to keep database consistency and to keep the reverting changes to the minimum needed.

May close (hopefully) https://github.com/woocommerce/woocommerce/issues/27419.

**Note for reviewers:** To check the code changes in the widget itself you may find it useful to open [the diff of this PR against 4.3.0](https://github.com/woocommerce/woocommerce/compare/4.3.0...revert-improved-filtering-for-variations), go to "Files changed" and search for `includes/widgets/class-wc-widget-layered-nav.php` (I didn't fully revert this one since there are small improvements that can be kept, e.g. for the code sniffer).

### How to test the changes in this Pull Request:

Basically, verify that the filtering widget works as in WooCommerce 4.3. This implies that even if "hide out of stock items" is set, there are simple and variable products, and at least one of them has stock, the widget will count all the attributes including:

* The ones used by simple products.
* The ones used by variable products, but not for variations.
* The ones used by variable products, for variations, even those not having stock.

### Other information:


### Changelog entry

> Fix - Revert the changes in filtering by attribute that were introduced in WooCommerce 4.4
